sidechain/test/src/sidechain_block_builder.rs

```diff
@@ -111,13 +113,27 @@ impl SidechainBlockBuilder {
 		self
 	}
 
+	/// Calculate the payload of a sidechain block.
+	pub fn block_data_hash(&self) -> H256 {
+		(
+			self.timestamp,
+			self.parentchain_block_hash,
+			self.signer.public(),
+			self.signed_top_hashes.as_slice(),
+			self.encrypted_payload.as_slice(),
+		)
+			.using_encoded(BlakeTwo256::hash)
```

Comment:
It would be nice, if we only had one implementation for this hash calculation. Otherwise, we have the risk of test code diverging from production code. Unless this will be changed anyway with the next PR?